DataFormats/NanoAOD/interface/FlatTable.h

```diff
@@ -161,25 +159,30 @@ namespace nanoaod {
 
   private:
     template <typename T>
-    typename std::vector<T>::const_iterator beginData(unsigned int column) const {
-      const Column &col = columns_[column];
-      check_type<T>(col.type);  // throws if type is wrong
-      return bigVector<T>().begin() + col.firstIndex;
+    auto beginData(unsigned int column) const {
+      return bigVector<T>().cbegin() + columns_[column].firstIndex;
     }
     template <typename T>
-    typename std::vector<T>::iterator beginData(unsigned int column) {
-      const Column &col = columns_[column];
-      check_type<T>(col.type);  // throws if type is wrong
-      return bigVector<T>().begin() + col.firstIndex;
+    auto beginData(unsigned int column) {
+      return bigVector<T>().begin() + columns_[column].firstIndex;
     }
 
     template <typename T>
-    const std::vector<T> &bigVector() const {
-      throw cms::Exception("unsupported type");
+    auto const &bigVector() const {
+      CMS_SA_ALLOW return const_cast<FlatTable *>(this)->bigVector<T>();
```

Comment:
oddly enough we are still getting 
```const_cast was used, this may result in thread-unsafe code```
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ce380e/7607/llvm-analysis/report-819fa2.html#EndPath

@makortel @Dr15Jones 
please advise if there is a solution to silence the SA warning

Reply:
I suspect the silencing of `const_cast` with `CMS_SA_ALLOW` has not made from an idea to the static analyzer. We could take another look in that, but for time being the best alternative is to add comment explaining why the `const_cast` is acceptable here.

Reply:
Okay! Should I still keep the `CMS_SA_ALLOW` anyway? It doesn't seem to be harmful, and if it one day also silences `const_cast` it is already there.

I can add a comment in the line before:
```
// justified use of `const_cast` to provide a `const` version of the `bigVector` accessor without code duplication
```

Reply:
I was just checking if the core guidelines have anything to say about this situation, and actually :+1: